foxai/explainer/computer_vision/algorithm/gradient_utils.py

```diff
@@ -0,0 +1,174 @@
+from inspect import signature
+from typing import Any, Callable, Optional, Tuple
+
+import torch
+
+from foxai.types import LayerType, ModelType, TargetType
+
+
+def _run_forward(
+    forward_func: Callable,
+    input_tensor: torch.Tensor,
+    target: Optional[TargetType] = None,
+    additional_forward_args: Optional[Tuple[Any]] = None,
+) -> torch.Tensor:
+    r"""Executes forward pass and returns result with respect to the target.
+
+    Args:
+        forward_func: forward function. This can be for example model's forward function.
+        input_tensor:      Input to be passed to forward_fn.
+        target: Index of the target class for which gradients must be computed (classification only).
+        additional_forward_args: Additional input arguments that forward function requires.
+    """
+    if len(signature(forward_func).parameters) == 0:
+        output = forward_func()
+    elif additional_forward_args is not None:
+        output = forward_func((input_tensor, *additional_forward_args))
+    else:
+        output = forward_func(input_tensor)
+
+    if target is None:
+        return output
+    elif isinstance(target, int):
+        return output[:, target]
+    elif isinstance(target, torch.Tensor):
```

Comment:
Do we really need to support `torch.Tensor` in this class? We don't use tensor operations here - only access value of tensor which imposes additional requirements/limitations that we need to check. This function will be much simpler if only `int` type will be accepted.

Reply:
On this point I totally agree although this will lead to incompatibility with targets accepted by other explainer methods. I think we could remove it after all methods are rewritten or if we think this is small issue I can improve signature for this algorithm only.  Wdyt?